test/sets.jl

```diff
@@ -147,6 +147,10 @@ s = union(Set([5,6,7,8]), Set([7,8,9]))
 s = Set([1,3,5,7])
 union!(s,(2,3,4,5))
 @test isequal(s,Set([1,2,3,4,5,7]))
+@test ===(typeof(union(Set([1]), IntSet())), Set{Int})
```

Comment:
Don't worry with existing code. You can change it too if you like (in a separate commit), but that's really not a requirement.